salt: remove loopback CNI plugin

alexandre-allard opened the pull request:
**Component**: salt

**Context**:
Since #2800, this SLS is no longer used anywhere except in node role, so let put everything that's inside it into the node role.

**Summary**:

**Acceptance criteria**:

alexandre-allard committed:
salt: flatten node-without-calico sls into node role

This SLS is no longer used anywhere except in node
role, so let put everything that's inside it into
the node role.

bert-e commented:
# Hello alexandre-allard-scality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @thomasdanan

* @alexandre-allard-scality

* @tcarmet

* @nootal

* @teddyandrieux

* @lucieleonard

* @gdemonet

* @sayf-eddine-scality

* @nicolast

* @jbertran

* @monpote

* @carlito-scality

* @slaperche-scality

* @cmonfort

alexandre-allard-scality commented:
/approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @thomasdanan

* @alexandre-allard-scality

* @tcarmet

* @nootal

* @teddyandrieux

* @lucieleonard

* @gdemonet

* @sayf-eddine-scality

* @nicolast

* @jbertran

* @monpote

* @carlito-scality

* @slaperche-scality

* @cmonfort










*The following options are set:* **approve**

gdemonet commented on salt/metalk8s/roles/node/init.sls:
```diff
@@ -1,3 +1,7 @@
 include:
-  - metalk8s.roles.internal.node-without-calico
+  - metalk8s.node.grains
+  - metalk8s.kubernetes.cni.loopback
   - metalk8s.kubernetes.cni.calico
+  - metalk8s.kubernetes.kubelet
+  - metalk8s.kubernetes.apiserver-proxy
+  - metalk8s.internal.preflight
```
Why the re-ordering, compared to what was done before?

alexandre-allard replied:
No real good reason, just to have both cni stuff grouped, since the ordering here does not seem to be important.

gdemonet replied:
Mmh, OK. And do we still need to include `cni.loopback` before `cni.calico`?

alexandre-allard-scality replied:
Do you mean, do we still need to include it or do we need to include it "before" ?

alexandre-allard replied:
I've completely removed the loopback plugin.

alexandre-allard committed:
salt: remove loopback CNI plugin

We no longer need this plugin since we now deploy
Calico on all nodes.

alexandre-allard committed:
salt: add formula to properly clean loopback CNI plugin

We ensure the configuration file for loopback
CNI plugin is removed when calling Calico formula.

bert-e commented:
# Branches have diverged



This pull request's source branch `improvement/flatten-internal-in-node-role` has diverged from
`development/2.6` by more than 50 commits.

To avoid any integration risks, please re-synchronize them using one of the
following solutions:

* Merge `origin/development/2.6` into `improvement/flatten-internal-in-node-role`
* Rebase `improvement/flatten-internal-in-node-role` onto `origin/development/2.6`

Note: If you choose to rebase, you may have to ask me to rebuild
integration branches using the `reset` command.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @thomasdanan

* @alexandre-allard-scality

* @tcarmet

* @nootal

* @teddyandrieux

* @lucieleonard

* @gdemonet

* @sayf-eddine-scality

* @nicolast

* @jbertran

* @monpote

* @carlito-scality

* @slaperche-scality

* @cmonfort

TeddyAndrieux commented on salt/metalk8s/orchestrate/upgrade/post.sls:
```diff
@@ -2,3 +2,4 @@
 
 include:
   - metalk8s.addons.prometheus-operator.post-upgrade
+  - metalk8s.kubernetes.cni.loopback.absent
```
No you cannot include it as is here, this state is run on the salt-master so if you do this it will just remove the cni in the salt master container (so likely do not remove anything)
So to me you need to create another orchestrate that run this on every minions
OR
(maybe better) Since we do not need this one once calico is configured as CNI, just include this absent in the calico state (with a comment to say we can remove it in 2.7)

TeddyAndrieux replied:
Or maybe we want to remove all others CNI plugins when we deploy calico :thinking: I don't know if it can cause trouble to have different CNI plugins
@NicolasT @gdemonet Any idea ?
(Anyway it's not the goal of this PR)

NicolasT replied:
Shouldn't cause 'trouble' since they're picked based on filename, hence the `10-` and `30-` (or something) prefix we use.

alexandre-allard replied:
Yeah, I saw during my test that the file was not properly removed, then I forgot to change it.
It's now done.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @nicolast

* @lucieleonard

* @alexandre-allard-scality

* @monpote

* @slaperche-scality

* @teddyandrieux

* @cmonfort

* @thomasdanan

* @carlito-scality

* @jbertran

* @sayf-eddine-scality

* @gdemonet

* @nootal

TeddyAndrieux reviewed (approved):
LGTM

alexandre-allard commented:
/approve

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/flatten-internal-in-node-role` into
`development/2.6`
* [w/2.7/improvement/flatten-internal-in-node-role](https://github.com/scality/metalk8s/compare/development/2.7...w/2.7/improvement/flatten-internal-in-node-role) will be merged into `development/2.7` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/7fc49b29f87ace037d2cec2fbd32b0e773da916e) did not succeed in branch improvement/flatten-internal-in-node-role.




*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/f694683f2ac62913f05b5bec239f8bcd2dc96a3a) did not succeed in branch w/2.7/improvement/flatten-internal-in-node-role.




*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/7fc49b29f87ace037d2cec2fbd32b0e773da916e) did not succeed in branch improvement/flatten-internal-in-node-role.




*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/f694683f2ac62913f05b5bec239f8bcd2dc96a3a) did not succeed in branch w/2.7/improvement/flatten-internal-in-node-role.




*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.6`

* :heavy_check_mark: `development/2.7`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.6`

* :heavy_check_mark: `development/2.7`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`



Please check the status of the associated issue None.

Goodbye alexandre-allard-scality.
